src/app/data_harvester.rs

```diff
@@ -288,17 +293,96 @@ impl DataCollector {
 
         self.update_cpu_usage();
         self.update_memory_usage();
-        self.update_processes();
         self.update_temps();
-        self.update_network_usage();
-        self.update_disks();
-
         #[cfg(feature = "battery")]
         self.update_batteries();
+        #[cfg(feature = "gpu")]
+        self.update_gpus(); // update_gpus before procs for gpu_pids but after temp/batteries/cpu_usage for appending
+        self.update_processes();
+        self.update_network_usage();
+        self.update_disks();
 
         // Update times for future reference.
         self.last_collection_time = self.data.collection_time;
     }
+    #[cfg(feature = "gpu")]
+    #[inline]
+    fn update_gpus(&mut self) {
```

Comment:
Not sure if I would prefer doing it like this or delegating out some of these parts to the individual functions that already handle them.

Reply:
Consolidation was to avoid iterating over the devices multiple times. ie the multiple `device_by_index` call loops that would be done for each collector. Might not be as significant if the power/util is removed.